src/kernels/jupyter/serverSelector.ts

```diff
@@ -104,6 +105,8 @@ export async function validateSelectJupyterURI(
             if (!handled) {
                 return DataScience.jupyterSelfCertExpiredErrorMessageOnly;
             }
+        } else if (err && err instanceof JupyterInvalidPasswordError) {
+            return DataScience.passwordFailure;
```

Comment:
This change is required to get a proper error message when password is inavlid
None of this code path was tested, hence none of these things work with the new kernel picker
Hence the need for more changes